subcommand/server-acl-init/command_ent_test.go

```diff
@@ -0,0 +1,436 @@
+	require.NoError(err)
+	for _, expected := range expectedPolicies {
+		found := false
+		for _, p := range policies {
+			if p.Name == expected {
+				found = true
+
+				// We assert that the policy doesn't have any namespace config
+				// in it because later that's what we're using to test that it
+				// got updated.
+				policy, _, err := consul.ACL().PolicyRead(p.ID, nil)
+				require.NoError(err)
+				require.NotContains(policy.Rules, "namespace")
+				break
+			}
+		}
+		require.True(found, "Did not find "+expected)
+	}
+
+	// Re-run the command with namespace flags. The policies should be updated.
+	cmd = Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	responseCode = cmd.Run(secondRunArgs)
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+	// Check that the policies have all been updated.
+	policies, _, err = consul.ACL().PolicyList(nil)
+	require.NoError(err)
+	for _, expected := range expectedPolicies {
+		found := false
+		for _, p := range policies {
+			if p.Name == expected {
+				found = true
+				policy, _, err := consul.ACL().PolicyRead(p.ID, nil)
+				require.NoError(err)
+				if expected == "connect-inject-token" {
+					// The connect inject token doesn't have namespace config,
+					// but does change to operator:write from an empty string.
+					require.Contains(policy.Rules, "operator = \"write\"")
+				} else {
+					require.Contains(policy.Rules, "namespace")
+				}
+				break
+			}
+		}
+		require.True(found, "Did not find "+expected)
+	}
+}
+
+// Test the auth method and acl binding rule created when namespaces are enabled
+// and there's a single consul destination namespace.
+func TestRun_ConnectInject_SingleDestinationNamespace(t *testing.T) {
+	t.Parallel()
+	k8s, testAgent := completeEnterpriseSetup(t, resourcePrefix)
+	defer testAgent.Stop()
+	setupK8sServiceAccount(t, k8s)
+	require := require.New(t)
+
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:        ui,
+		clientset: k8s,
+	}
+	cmd.init()
+	args := []string{
+		"-server-label-selector=component=server,app=consul,release=" + releaseName,
+		"-resource-prefix=" + resourcePrefix,
+		"-k8s-namespace=" + ns,
+		"-expected-replicas=1",
+		"-create-inject-auth-method",
+		"-enable-namespaces",
+		"-consul-inject-destination-namespace=dest",
+	}
+
+	responseCode := cmd.Run(args)
+	require.Equal(0, responseCode, ui.ErrorWriter.String())
+
+	bootToken := getBootToken(t, k8s, resourcePrefix)
+	consul, err := api.NewClient(&api.Config{
+		Address: testAgent.HTTPAddr,
+		Token:   bootToken,
+	})
+	require.NoError(err)
+
+	// Check the ACL auth method is created in the expected namespace.
+	authMethodName := "release-name-consul-k8s-auth-method"
+	methods, _, err := consul.ACL().AuthMethodList(&api.QueryOptions{
+		Namespace: "dest",
+	})
+	require.NoError(err)
+	found := false
+	for _, m := range methods {
+		if m.Name == authMethodName {
+			found = true
+		}
+	}
+	require.True(found, fmt.Sprintf("no auth method with name %s found", authMethodName))
```

Comment:
There should be some validation about the properties of the auth method here, since they matter and can vary with namespace enabled.

Specifically, it would be good to check its Config: `MapNamespaces` and `ConsulNamespacePrefix` fields.

Reply:
Yep, done!